specification/dsdl/serializable_types.tex

```diff
@@ -0,0 +1,285 @@
+
+\subsubsection{Dependencies}
+
+In order to refer to a composite type from another composite type definition
+(e.g., for nesting or for referring to an external constant),
+one has to specify the full data type name of the referred data type followed by its
+major and minor version numbers separated by the namespace separator character,
+as demonstrated on the figure \ref{fig:dsdl_nested_reference}.
+
+To facilitate look-up of external dependencies,
+implementations are expected to obtain from external sources%
+\footnote{For example, from user-provided configuration options.}
+the list of directories that are the roots of namespaces containing the referred dependencies.
+
+\begin{figure}[H]
+    $$
+    \underbrace{\texttt{\huge{uavcan.node.Heartbeat}}}_{\text{full data type name}}%
+    \texttt{\huge{.}}%
+    \underbrace{\texttt{\huge{1.0}}}_{\substack{\text{version} \\ \text{numbers}}}%
+    $$
+    \caption{Reference to an external composite data type definition.\label{fig:dsdl_nested_reference}}
+\end{figure}
+
+If the referred data type and the referring data type share the same full namespace name,
+it is allowed to omit the namespace from the referred data type specification
+leaving only the short data type name, as demonstrated on the figure \ref{fig:dsdl_nested_reference_short}.
+In this case, the referred data type will be looked for in the namespace of the referrer.
+Partial omission of namespace components is not permitted.
+
+\begin{figure}[H]
+    $$
+    \underbrace{\texttt{\huge{Heartbeat}}}_{\text{short data type name}}%
+    \texttt{\huge{.}}%
+    \underbrace{\texttt{\huge{1.0}}}_{\substack{\text{version} \\ \text{numbers}}}%
+    $$
+    \caption{Reference to an external composite data type definition located in the same namespace.
+    \label{fig:dsdl_nested_reference_short}}
+\end{figure}
+
+Circular dependencies are not permitted.
+A circular dependency renders all of the definitions involved in the dependency loop invalid.
+
+If any of the referred definitions are marked as deprecated,
+the referring definition must be marked deprecated as well\footnote{%
+Deprecation is indicated with the help of a special directive, as explained in section \ref{sec:dsdl_directives}}.
+If a non-deprecated definition refers to a deprecated definition,
+the referring definition is malformed\footnote{%
+This tainting behavior is designed to prevent unexpected breakage of
+type hierarchies when one of the deprecated dependencies reaches its end of life.}.
+
+\subsubsection{Unions}\label{sec:dsdl_composite_tagged_unions}
+
+Any data schema definition can be supplied with a special directive (section \ref{sec:dsdl_directives})
+indicating that it forms a tagged union.
+
+A tagged union schema must contain at least two field attributes.
+A tagged union must not contain padding field attributes.
+
+The value of a tagged union object is a function of the field attribute which value it is currently holding
```

Comment:
This sentence is missing commas but should be revised per my previous  comments about  tagged union  value functions.